state/protocol/badger/mutator.go

```diff
@@ -235,6 +239,10 @@ func (m *Mutator) Extend(candidate *flow.Block) error {
 	// check if the receipts in the payload are valid
 	err = m.receiptExtend(candidate)
 	if err != nil {
+		if state.IsInvalidExtensionError(err) {
+			return err
+		}
+
```

Comment:
not sure this is needed. By wrapping
https://github.com/onflow/flow-go/blob/c5671a21f466bcef93fd105277ee827f490f581c/state/protocol/badger/mutator.go#L246
the error type is preserved.

For example:
```golang
        err0 := NewInvalidExtensionError(msg string)
        err1 := fmt.Errorf("payload receipts not compliant with chain state: %w", err0)
 
        state.IsInvalidExtensionError(err0) // true
        state.IsInvalidExtensionError(err1) // true
```

In fact, it is good practise to wrap errors to make expose information about the error trace. In other words, even if `err` is an `InvalidExtensionError`, it would still be preferred to wrap the error here:
```golang
return fmt.Errorf("payload receipts not compliant with chain state: %w", err0)
```

I am sure you have reasons for wrapping the error / not wrapping the error depending on its type. Maybe my [earlier suggestion](https://github.com/onflow/flow-go/pull/180#discussion_r531259144) would also remove the need to not wrap the error here?